### ISSUE TITLE
Save file corruption issue

### ISSUE THREAD

**Issue by eforgacs** [sithlord48/blackchocobo#50]
Hi there,

I'm an occasional contributor to Black Chocobo. I was contacted recently by someone who wanted to submit a bug report. He said his save file got corrupted. Here are the details he sent me (I apologize in advance for the wall of text - just passing it along in case it helps someone):

"i am currently playing ff7 on pc by using the save editor. there are bugs in the game and my save file became corrupted and i am unable to continue the game. i am playing ff7 on steam by using the reunion mod. i installed r06k and the gamma 3. i put the syw mod in the reunion folder and i enabled character enhancements. i will tell you exactly how i used the saved game editor. in 3 or 4 points during the game i increased the money by using the saved game editor and then i spent the money. also i didnt feel like grinding to raise the materia level so i gradually raised the levels at certain points during the game. after i got cloud back after he fell into the lifestream i used the editor to get the needed chocobos for mating, i bred the gold chocobo and then i edited the gold chocobo stats. that's all what i did. also i edited the item quantities a few times. everything was normal until the space mission. i got into the escape pod, it landed and they are in the highwind talking. after red 13 says lets go to cosmo canyon and i told the pilot to take off you will be flying near rocket town. for me when i spoke to the pilot and i told him to take off i was suddenly in front of the north crater and i crashed into it and cid said that it's powerful. this is the first bug. i continued playing the game.  i got the key of the ancients, i went to the forgotten city, i saw the cutscene of aerith and the materia falling from her hair, i got the phone call saying that the cannon has been moved and then i left the forgotten city. if i examine highwind then cloud should feel an earthquake, weapon should come out of the ocean and star walking towards midgar and then i will get a cutscene in the airship and cait sith is beeping.
for me nothing happened when i examined the highwind and it took off. if i press the menu button then i get the cutscene which automatically comes after the earthquake in the cockpit and then the view goes back to the world map. the world map music is playing and nothing is happening. i am unable to continue the game because the weapon didn't appear
weapon is not attacking and i cant progress the game. i asked someone to try my saved file. the same thing happened. i also tested the saved file on my other laptop which has a fresh install of ff7 without the reunion mod. the same thing happened. i made a complaint with the reunion team and i told them the issue. i told them that i used black chocobo saved game editor. they said that the save editor corrupted the saved game. i suspected that it's a game glitch. if you save the game while the weapon is walking and you reload it then weapon vanishes and battle music is playing in the background and nothing is happening. that has happened with other people. it doesn't look like this game bug caused the issue because weapon vanished for me before it appeared."

That's what he sent me. Hope this helps you guys out in some way. Best of luck.

**Comment by sithlord48** [comment]
Any of those mods could have bugs that cause this alone or in combination. 

Quick read it sounds to me like they changed their world location, or maybe a field item that one of those mods uses as another flag. They could have changed a progression variable that one of the mods changes. It could literally be anything.  

Without a set of saves there is not much i can do for them.

**Comment by eforgacs** [comment]
Gotcha. Thanks! Just figured I'd pass it along.

**Comment by cns00** [comment]
hi,

i am the person who has the corrupted save file

> Quick read it sounds to me like they changed their world location, or maybe a field item that one of those mods uses as another flag. They could have changed a progression variable that one of the mods changes. It could literally be anything.

it's not the fault of the reunion mod. if i had played the game with the reunion mod and without using the black chocobo save game editor then highwind will be next to rocket town when i tell the pilot to take off after i finished the space mission part. also diamond weapon will appear on the world map when i examine highwind after i exit to the world map after bugenhagen puts the key of the ancients into the device in the forgotten city and i see the cutscene of materia falling from aerith's hair 

my save file is clearly corrupted because diamond weapon didn't appear on the world map when i examined highwind and it just took off. also i get the scene in the cockpit that shows after diamond weapon appears on the world map if i press the menu button and then the view changes to to world map. that happens with and without the reunion mod

this is my save file:-
[save00.zip](https://github.com/sithlord48/blackchocobo/files/9817447/save00.zip)

the save file has 3 slots:-
1) on the world map next to the forgotten city. you need to go in and go to the fountain and see the cutscene of aerith
2) in the submarine just before the part which you have to sink the red submarine
3) in a house in the forgotten city. it's after the events at the fountain and you have to go to the world map and enter highwind

please load the 3rd slot and exit to the world map and examine highwind

**Comment by sithlord48** [comment]
do you have the save before you changed anything? because without it i can't help you . Again not that i might be able to anyway its likely a bug caused by combining mods and or editing stuff you should not have touched.

**Comment by cns00** [comment]
this is not the first time that i played ff7 on steam. i have played it a bunch of times in the past years and i have used black chocobo saved game editor. there were no issues. this is the first time that i played the game by using the reunion mod and the save editor. i don't know. it looks like the saved files that the saved game editor makes are not perfectly compatible with mods

my save file is corrupted so i am unable to continue the game by using it so i had to download an ff7 saved game pack and use a saved file from the pack. you can download the saved game pack from https://drive.google.com/file/d/0B0czBXR0PqOMaG4xZ1EtYnRSMmk2QXpRS1dlWWVzdw/view?resourcekey=0-CB0w4wyAPKQ0pWG4NZ_E-A. i used save07.ff7 and i loaded the second forgotten capital save slot and i continued from there

please examine the coding of my corrupted save file and see if you can find anything wrong in the coding

**Comment by sithlord48** [comment]
>it looks like the saved files that the saved game editor makes are not perfectly compatible with mods

This depends on if a mod changes anything in the savemap. If you edit the save (with any editor) you could have problems. There is no way for me(or anyother tool maker)  to know what (if any) changes they make to the save map for any given mod. For this reason you should always keep a back up when editing your saves.

> please examine the coding of my corrupted save file and see if you can find anything wrong in the coding

Finding out what is wrong with your broken save does not help me fix any potential bug. Unless you provide me with the save before you edited it there is no way for me to figure out what changes (again if any) were made incorrectly to the save. 

TLDR: Lots of care it taken to ensure that only the correct parts are changed. The object that is touching your savefile has [extensive tests](https://github.com/sithlord48/ff7tk/blob/master/unittests/data/FF7Save_test.h) to prove that all the data is being written correctly. If any test fails the build fails and nothing new is released.

**Comment by cns00** [comment]
unfortunately i do not have a backup save file which was made before i used
the save game editor. i used the saved game editor at regular intervals in
the game and all was ok when i loaded the saved game. the bugs happened
later in the game

you said that you need a save file before the save editor was used to
compare. i will tell you how to get it. download the saved game pack, find
the save file that i told you to use and load the save slot that i used to
continue the game. take screenshots of the character status screens, the
inventory screen, the materia list screen and the limit break screens.
after that load my corrupted save file in the save editor, pick the 3rd
slot and edit the character info so it matches the character info in the
screenshots. after that edit the inventory list and the materia list so it
matches the screenshots. try to make the 3rd slot in my corrupted save file
to be as close as possible to the save slot in the save file that you got
from the save file pack. finally load the modified 3rd slot in the game,
change the party members and their equipment so that it matches the party
the party to which is in save slot in the save file that you got from the
save file pack and then save again. the save file should still be corrupted
and highwind will take off instead of diamond weapon showing. now you will
be able to compare my corrupted save file to the other save file that you
got from the save file pack and that save file will be before the saved
game editor was used
